distribution/lib/Standard/AWS/0.0.0-dev/src/Errors.enso

```diff
@@ -38,6 +40,18 @@ type S3_Bucket_Not_Found
     to_display_text : Text
     to_display_text self = "Bucket '" + self.bucket + "' not found."
 
+## An error when a S3 key is not found within a bucket.
+
+   It is a low-level error used by the raw methods.
+   `S3_File` will translate this error to a `File_Error.Not_Found`.
+type S3_Key_Not_Found
+    ## PRIVATE
+    Error bucket:Text key:Text (credentials : AWS_Credential | Nothing)
```

Comment:
I also didn't like it too much.

My reason for having it is that this is the 'lower layer' error used in raw `S3`. In `S3_File` it will be converted to a `File_Error.Not_Found` to ensure consistent error types of our file backends. That error needs a `File_Like` - in this case the `S3_File`, so I figured we need to be able to reconstruct it from this error.

I guess I could try to do it differently and pass the `S3_File` to the handler that is converting the error. I was only worried that the handler may be used in contexts where there are multiple files being manipulated and it will not be able to tell _which_ file was the missing one. That's why giving enough info closer to the error source seemed like a better approach, but maybe it isn't after all.

Reply:
Fixed. Now the credentials are added only in `S3_File` in `translate_file_errors` to allow to reconstruct the `S3_File`.